### PR TITLE
remove Offer, Security and SecurityStorage modules, update to Move 1.2, move to Diem standard library

### PR DESCRIPTION
**Pull request by mkurnikov**

### CODE REVIEW

File: modules/pontem.move [hunk 1]
```diff
@@ -25,71 +25,77 @@ module Pontem {
...
     }
 
-    public fun withdraw<Coin>(coin: &mut T<Coin>, amount: u128): T<Coin> {
+    public fun withdraw<Coin: store>(coin: &mut T<Coin>, amount: u128): T<Coin> {
         assert(coin.value >= amount, ERR_CANT_WITHDRAW);
         coin.value = coin.value - amount;
         T { value: amount }
     }
 
+    native public fun deposit_native<Token: store>(address: &signer, amount: u128): Pontem::T<Token>;
```
Comment (borispovod) [inline modules/pontem.move:66]: Just T<Coin>?

Comment (borispovod) [reply modules/pontem.move:66]: `T<Coin>`

File: modules/pontem.move [hunk 2]
```diff
@@ -25,71 +25,77 @@ module Pontem {
...
+
+    native public fun withdraw_native<Token: store>(address: &signer, balance: Pontem::T<Token>);
```
Comment (borispovod) [inline modules/pontem.move:68]: Same here?

**Review by borispovod — CHANGES_REQUESTED** [review]
Need fixes, see my comments.
Also, we need to adopt new Diem's standard library instead of using our own, but for now it's ok.

File: modules/Event.move [hunk 3]
```diff
@@ -0,0 +1,86 @@
+address 0x1 {
+
+/// The Event module defines an `EventHandleGenerator` that is used to create
+/// `EventHandle`s with unique GUIDs. It contains a counter for the number
+/// of `EventHandle`s it generates. An `EventHandle` is used to count the number of
+/// events emitted to a handle and emit events to the event store.
+module Event {
```
Comment (borispovod) [inline modules/Event.move:7]: We have custom event logic in Pontem, does @RIg410 know about event changes?

**Review by borispovod — CHANGES_REQUESTED** [review]
Look at events please, otherwise seems fine. 
And run pipelines with new dove version, so it will pass tests.

File: old_modules/u256.move [hunk 4]
```diff
@@ -0,0 +1,37 @@
+address 0x1 {
```
Comment (borispovod) [inline old_modules/u256.move:1]: why needs old_modules?

File: modules/Event.move (hunk 3 above)
Comment (mkurnikov) [reply modules/Event.move:7]: I asked him, he said that we support both versions, so I've added one from Diem. Should I change it into our simpler one?

File: modules/Сoins.move [hunk 5]
```diff
@@ -0,0 +1,23 @@
+address 0x1 {
+
+/// Deprecated!!!!
```
Comment (fzzr-) [inline modules/Сoins.move:3]: Explain the reasons for deprecation please.

**Review by fzzr- — APPROVED** [review]

**Comment by fzzr-** [comment]
@borispovod [This](https://github.com/pontem-network/move-stdlib/pull/12#pullrequestreview-668456206) is resolved now but unfortunately I can't mark it as resolved. So I'll merge it now.

**Pull request merged**